README.md

```diff
@@ -1237,10 +1237,101 @@ class MyClass
 
 ### PhpDoc on methods
 
-We put PhpDoc comment on all methods with exception of constructors (can be skipped in some cases).
+#### No PhpDoc on fully strictly typed methods
 
-We put PhpDoc comment on constructors when IDE (for example PhpStorm) cannot guess classes of attributes,
-return type etc. It’s optional otherwise. For example, if we inject some scalar type, we must put PhpDoc comment.
+We put a PhpDoc comment on methods where the return and argument types cannot be determined by the IDE. For example, 
+prior to PHP 7.0, we would need a PhpDoc to type-hint scalar types. In PHP 7, the scalar types would be strictly 
+typed and PhpDoc is not needed.
+
```

Comment:
I think we should also add clause when we do not put any comment. Currently we just require to put it when it's needed, but not forbid to put it when it's not needed at all. With such clause we should note that we can always use PhpDoc if we add any additional information besides what's already known by definition